Source/JavaScriptCore/wasm/js/WebAssemblyTablePrototype.cpp

```diff
@@ -155,35 +149,10 @@ JSC_DEFINE_HOST_FUNCTION(webAssemblyTableProtoFuncSet, (JSGlobalObject* globalOb
             return throwVMTypeError(globalObject, throwScope, "WebAssembly.Table.prototype.set requires the second argument for non-defaultable table type"_s);
     }
 
-    if (table->table()->asFuncrefTable()) {
-        WebAssemblyFunction* wasmFunction = nullptr;
-        WebAssemblyWrapperFunction* wasmWrapperFunction = nullptr;
-        if (!value.isNull() && !isWebAssemblyHostFunction(value, wasmFunction, wasmWrapperFunction))
-            return throwVMTypeError(globalObject, throwScope, "WebAssembly.Table.prototype.set expects the second argument to be null or an instance of WebAssembly.Function"_s);
-
-        if (value.isNull())
-            table->clear(index);
-        else {
-            ASSERT(value.isObject() && isWebAssemblyHostFunction(jsCast<JSObject*>(value), wasmFunction, wasmWrapperFunction));
-            ASSERT(!!wasmFunction || !!wasmWrapperFunction);
-            if (wasmFunction)
-                table->set(index, wasmFunction);
-            else
-                table->set(index, wasmWrapperFunction);
-        }
-    } else {
-        if (isExternref(table->table()->wasmType()))
-            table->set(index, value);
-        else {
-            RELEASE_ASSERT(Options::useWasmGC());
-            JSValue internalValue = Wasm::internalizeExternref(value);
-            if (!Wasm::TypeInformation::castReference(internalValue, true, table->table()->wasmType().index))
-                return throwVMTypeError(globalObject, throwScope, "WebAssembly.Table.prototype.set failed to cast the second argument to the table's element type"_s);
-            table->set(index, internalValue);
-        }
-    }
+    table->set(globalObject, index, value);
+    RETURN_IF_EXCEPTION(throwScope, encodedJSUndefined());
 
```

Comment:
Let's do,
```
scope.release();
table->set(globalObject, index, value);
return JSValue::encode(jsUndefined());
```

Reply:
Yeah, I thought about that, I wasn't sure it was clearer and could have bugs if people add code between the call and return. I'll change it though.